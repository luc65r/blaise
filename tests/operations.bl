PROGRAMME Test
VARIABLES
    n : Entier
DÉBUT
    n <- 12 * 4322127149856432
    n <- 6 * n + 214 * 65169 DIV 15 - 192513 + n
    écrire(n)
FIN
