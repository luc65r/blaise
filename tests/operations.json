{
  "loc": "1,1-8,4",
  "name": "Test",
  "subs": [],
  "vars": [
    {
      "loc": "3,5-3,15",
      "var": {
        "loc": "3,5-3,6",
        "name": "n"
      },
      "type": {
        "loc": "3,9-3,15",
        "name": "Entier"
      }
    }
  ],
  "stmts": [
    {
      "loc": "5,5-5,31",
      "lval": {
        "loc": "5,5-5,6",
        "var": {
          "loc": "5,5-5,6",
          "name": "n"
        }
      },
      "rval": {
        "loc": "5,10-5,31",
        "binexpr": {
          "loc": "5,10-5,31",
          "op": "*",
          "left": {
            "loc": "5,10-5,12",
            "intlit": "12"
          },
          "right": {
            "loc": "5,15-5,31",
            "intlit": "4322127149856432"
          }
        }
      }
    },
    {
      "loc": "6,5-6,49",
      "lval": {
        "loc": "6,5-6,6",
        "var": {
          "loc": "6,5-6,6",
          "name": "n"
        }
      },
      "rval": {
        "loc": "6,10-6,49",
        "binexpr": {
          "loc": "6,10-6,49",
          "op": "+",
          "left": {
            "loc": "6,10-6,45",
            "binexpr": {
              "loc": "6,10-6,45",
              "op": "-",
              "left": {
                "loc": "6,10-6,36",
                "binexpr": {
                  "loc": "6,10-6,36",
                  "op": "+",
                  "left": {
                    "loc": "6,10-6,15",
                    "binexpr": {
                      "loc": "6,10-6,15",
                      "op": "*",
                      "left": {
                        "loc": "6,10-6,11",
                        "intlit": "6"
                      },
                      "right": {
                        "loc": "6,14-6,15",
                        "var": {
                          "loc": "6,14-6,15",
                          "name": "n"
                        }
                      }
                    }
                  },
                  "right": {
                    "loc": "6,18-6,36",
                    "binexpr": {
                      "loc": "6,18-6,36",
                      "op": "div",
                      "left": {
                        "loc": "6,18-6,29",
                        "binexpr": {
                          "loc": "6,18-6,29",
                          "op": "*",
                          "left": {
                            "loc": "6,18-6,21",
                            "intlit": "214"
                          },
                          "right": {
                            "loc": "6,24-6,29",
                            "intlit": "65169"
                          }
                        }
                      },
                      "right": {
                        "loc": "6,34-6,36",
                        "intlit": "15"
                      }
                    }
                  }
                }
              },
              "right": {
                "loc": "6,39-6,45",
                "intlit": "192513"
              }
            }
          },
          "right": {
            "loc": "6,48-6,49",
            "var": {
              "loc": "6,48-6,49",
              "name": "n"
            }
          }
        }
      }
    },
    {
      "loc": "7,5-7,49",
      "expr": {
        "loc": "7,5-7,49",
        "subname": "Ã©crire",
        "args": [
          {
            "loc": "7,45-7,48",
            "var": {
              "loc": "7,45-7,48",
              "name": "n"
            }
          }
        ]
      }
    }
  ]
}
