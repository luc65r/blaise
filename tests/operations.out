363058680588677519
